### PR TITLE
Add assignation options

### PR DESCRIPTION
**Pull request by jocgir**
- Deprecated the old assignment syntax `$var := value`. Must use `@{var} : = value` or `@$var := value`
- Add support for assignment operators such as `@a += value`. All common operators are supported
- Add a special operator `@a ~= value` that could be used to assign variable if we don't know if it exists or not (i.e. a variable exist because it has been defined externally)
- Support local variable sub element assignation `@a.b := value` was correctly handled, but `@{a.b} := value` or `@$a.b := value` were not handled properly.

### CODE REVIEW

**Comment by julienduchesne** [comment]
I understand that this is how it's done in Go but Go is compiled and we have an IDE. Developing Gotemplate is really not as easy. Is there a reason to do this other than making it the same as Go?

**Comment by jocgir** [comment]
In fact, I just implemented a way to solve the errors we get if a variable exists (and we have no way to know that if it exists or not (i.e. disabling the assignation validation with `~=`).  The implementation of assignment operators then becomes easy to do and it may simplify the writing of go template code. Especially when appending elements to a list.

**Comment by julienduchesne** [comment]
Why not ditch the difference between all of these operators and never validate?

**Comment by jocgir** [comment]
The validation was added to be coherent with local variable assignment rule. It is useful to avoid unwanted override.

File: docs_tests/basic_features/assignation.md [hunk 1]
```diff
@@ -3,29 +3,67 @@
...
-| `@numeric4 := 5E-3;`                        | `{{- set $ "numeric4" 5E-3 }}`                           | Global assignation of small scientific notation number
-| `@hexa1 := 0x100;`                          | `{{- set $ "hexa1" 0x100 }}`                             | Global assignation of hexadecimal number
-| `@result1 := (2+3)*4;`                      | `{{- set $ "result1" (mul (add 2 3) 4) }}`               | Global assignation of mathematic expression
-| `@result2 := String("hello world!").Title;` | `{{- set $ "result2" ((String "hello world!").Title) }}` | Global assignation of generic expression
+As in Go lang, you must initially declare your global variable using the `:=` assignment operator and subsequent overwrite use the `=` operator.
+
+| Razor expression                            | Go Template                                                                     | Note
+| ----------------                            | -----------                                                                     | ----
+| `@string := "string value";`                | `<pre-assign check code>{{- set $ "string" "string value" }}`                   | Global declare and assign of string
```
Comment (julienduchesne) [inline docs_tests/basic_features/assignation.md]: I think having `<pre-assign check code>` in the doc would be confusing to an end-user. I think a note explaining that Razor adds a check before assigning (outside of the code example) would be more appropriate. I see code examples as `how can I do this` more than `what happens when I do this`

**Pull request merged**